File: src/main/java/LoadingWindow.java
package CurrentWeather;
import javax.swing.*;
import java.awt.*;

public class LoadingWindow
{
    private JFrame frame;

    public LoadingWindow() {
        frame = new JFrame();
        frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
        frame.setTitle("JWeather");
        frame.setSize(250, 250);
        frame.add(new JLabel("Fetching Weather..."), BorderLayout.CENTER);
        frame.setVisible(true);
    }

    public void dispose(){
        System.out.println("Disposing frame");
        frame.dispose();
    }
}

File: src/main/java/LoadingWindow.kt
package CurrentWeather

import java.awt.BorderLayout
import javax.swing.JFrame
import javax.swing.JLabel

class LoadingWindow : Runnable {
    private var frame: JFrame? = null

     override fun run() {
        frame = JFrame()
        frame!!.defaultCloseOperation = JFrame.DISPOSE_ON_CLOSE
        frame!!.title = "JWeather"
        frame!!.setSize(250, 250)
        frame!!.add(JLabel("Fetching Weather..."), BorderLayout.CENTER)
        frame!!.isVisible = true
    }

    fun dispose() {
        println("Disposing frame")
        frame!!.dispose()
    }
}

